exercicios/para-casa/sistema_de_alunas_patricia.py

```diff
@@ -0,0 +1,134 @@
+                                    "5 - Consultar status de aprovação\n"
+                                    "6 - Sair do sistema\n"
+                                    "Opção: "))
+                
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 5).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+            
+        return codigo_opcao
+    
+def incluir_nova_aluna():
+    #FUNÇÃO PARA INCLUIR NOVA ALUNA NO DATASET
+
+    nome = input ("Nome da nova aluna:")
+    sobrenome = input ("Sobrenome nova aluna:")
+    turma = input ("Turma:")
+    notas = list(map(float,input("Notas (separadas por espaço):").split()))
+    presencas = list(map(bool,map(int,input("Presenças (1 para presente,0 para ausente, separada por espaço):").split())))
+    participaçao = float(input("Nota de participação:"))    
+    
+    #valida os dados da entrada
+    if not all(0 <= nota <= 10 for nota in notas):
+        print("Erro:Presenças devem ser 1 (prensente)ou 0(ausente).")
+        return
+    if not all(presenca in [True,False] for presenca in presencas):
+        return
+    if not 0 <= participaçao <= 10:
+        print("Erro: Nota de participação deve estar no intervalo de 0 a 10. ")
+        return
+    
+# CRIA UM DICIONÁRIO COM DADOS DA NOVA ALUNA
+    nova_aluna ={
+        "nome": nome,
+        "sobrenome":sobrenome,
+        "turma": turma,
+        "notas": notas,
+        "presencas": presencas,
+        "participacao": participaçao
+
+    }
+
+    #adiciona a nova aluna ao dataset
+    dataset.append(nova_aluna)
+    print(f"Aluna{nome}{sobrenome}incluida com sucesso!")
+    
+
+
+
+def consultar_lista_alunas():
+    #função para consultar a lista de alunas cadastradas
+    if not dataset:
+        print("Não há alunas cadastrada")
+        return
+    print("\nLista de Alunas:")
+    #exibe o nome completo de cada aluna cadastrada
+    for aluna in dataset:
+         print(f"{aluna["nome"]} {aluna["sobrenome"]}")
+         
+
+
+    
+def consultar_faltas_aluna():
+     #função para consultar a quantidade de faltas de uma aluna especifica
+     nome_completo = input ("Nome completo da aluna: ") 
+     aluna = next((a for a in dataset if f"{a['nome']} {a["sobrenome"]}"  == nome_completo), None)   
+     if aluna:
+         faltas = aluna["presencas"].count(False)
+         print(f"Faltas da aluna {nome_completo}:{faltas}")
+     else:
+         print(f"Aluna {nome_completo} não encontrado.")    
+    
+
+
+def consultar_notas_aluna():
+    #função para consultar as notas de uma aluna especifica
+    nome_completo = input ("Nome completo da aluna:")
+    aluna = next((a for a in dataset if f"{a["nome"]}  {a["sobrenome"]}"== nome_completo), None)
+    if aluna:
+        print(f"Notas da aluna {nome_completo}: {aluna["notas"]}")
+    else:
+        print(f"Aluna {nome_completo} não encontrado")    
```

Comment:
Como não foi utilizado a tupla como chave do dicionário foi necessário implementar a função next. O que acabou gerando uma complexidade no código que não era necessário. No nosso caso era necessário apenas consultar pela chave.

**Sugestão:**
```suggestion
    nome = input('Informe o primeiro nome da aluna: ')
    sobrenome = input('Informe o sobrenome da aluna: ')
    chave = (nome, sobrenome)
    if chave in dataset:
        notas = dataset[chave]["Notas"]
        print(f'A aluna {nome} {sobrenome} está com as notas: {notas}')
    else:
        print('Aluna não encontrada.')
```